src/hooks/useReferredBy.ts

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { log } from '@/lib/telemetry';
+
+import { useAccounts } from './useAccounts';
+import { useDydxClient } from './useDydxClient';
+
+export const useReferredBy = () => {
+  const { dydxAddress } = useAccounts();
+  const { getReferredBy } = useDydxClient();
+
+  const queryFn = async () => {
```

Comment:
i think its more correct to get the arguments of queryFn (dydxAddress) here from the queryKeys that react-query passes in instead of from the outer React context - this is because dydxAddress can change in React, but the result of queryFn should be atomic with whatever queryKey is set internally in react-query. this is esp true if there is debounce or polling interval involved

Reply:
see here! https://tanstack.com/query/v4/docs/framework/react/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key